main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -538,27 +555,73 @@ object Lowering {
       val ms = methods.map(visitJvmMethod)
       Expression.NewObject(name, clazz, t, pur, eff, ms, loc)
 
+    // New channel expressions are rewritten as follows:
+    //     chan Int32 10
+    // becomes:
+    //     newChannel[Int32](10)
+    //
     case Expression.NewChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
       val t = visitType(tpe)
-      Expression.NewChannel(e, t, pur, eff, loc)
+      val newChannel = Expression.Def(Defs.ChannelNew, Type.mkImpureArrow(e.tpe, t, loc), loc)
+      Expression.Apply(newChannel, e :: Nil, t, pur, eff, loc)
 
+    // Channel get expressions are rewritten as follows:
+    //     <- c
+    // becomes:
+    //     get(c)
     case Expression.GetChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
       val t = visitType(tpe)
-      Expression.GetChannel(e, t, pur, eff, loc)
-
-    case Expression.PutChannel(exp1, exp2, tpe, pur, eff, loc) =>
+      val getChannel = Expression.Def(Defs.ChannelGet, Type.mkImpureArrow(e.tpe, t, loc), loc)
+      Expression.Apply(getChannel, e :: Nil, t, pur, eff, loc)
+
+    // Channel put expressions are rewritten as follows:
+    //     c <- 42
+    // becomes:
+    //     put(42, c)
+    case Expression.PutChannel(exp1, exp2, _, pur, eff, loc) =>
       val e1 = visitExp(exp1)
       val e2 = visitExp(exp2)
-      val t = visitType(tpe)
-      Expression.PutChannel(e1, e2, t, pur, eff, loc)
-
+      val putChannel = Expression.Def(Defs.ChannelPut, Type.mkImpureUncurriedArrow(List(e2.tpe, e1.tpe), Type.Unit, loc), loc)
+      Expression.Apply(putChannel, List(e2, e1), Type.Unit, pur, eff, loc)
+
+    // Channel select expressions are rewritten as follows:
+    //     select {
+    //         case x <- ?ch1 => ?handlech1
+    //         case y <- ?ch2 => ?handlech2
+    //         case _ => ?default
+    //     }
+    // becomes:
+    //     let ch1 = ?ch1;
+    //     let ch2 = ?ch2;
+    //     match selectFrom([mpmcAdmin(ch1), mpmcAdmin(ch2)]) @ Static, false) {  // true if no default
+    //         case (0, locks) =>
+    //             let x = unsafeGetAndUnlock(ch1, locks);
+    //             ?handlech1
+    //         case (1, locks) =>
+    //             let y = unsafeGetAndUnlock(ch2, locks);
+    //             ?handlech2
+    //         case (-1, _) =>                                                  // omitted if no default
+    //             ?default                                                     //
```

Comment:
Maybe add a comment that unlock is handled by selectFrom?

(Separately: I find this a strange design decision, but maybe it is required?)

Reply:
its to avoid unlockLocks as a public function

Reply:
OK, but in Lowering we dont have to respect visibility any more. But yeah, IDK.

Reply:
Oh, so everything in Channel could be private?

Reply:
Yes, expect the Redundancy phase will complain :D (Cant win, cant ya :))

Reply:
Ah, just mark it as `@test`. 

Actually why doesn't `@Internal` just make it okay to be unused like `@test`?

Reply:
It could.

Reply:
We could also add `@Unused`